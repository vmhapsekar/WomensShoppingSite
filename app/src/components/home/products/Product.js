import React from 'react'
import PropTypes from 'prop-types'
import {IMAGES_URL} from '../../../constants/appUrls'
import {ProductStatus, ProductName, ProductPrice, ProductImage, ProductWrapper} from './style'
import {Grid, GRow, GCol} from '../../common/grid/Grid'

const Product = ({image, name, price, isExclusive, isSale}) => {
  return (
    <ProductWrapper>
      <Grid fluid>
        <GRow>
          <GCol xs={12}>
            <ProductImage src={`${IMAGES_URL}${image}`} />
          </GCol>
        </GRow>
        <GRow>
          <GCol xs={12} sm={12} md={5} lg={5}>
            <ProductStatus isExclusive={isExclusive} isSale={isSale}>
              {isExclusive ? 'Exclusive' : ''}
              {isSale ? 'Sale' : ''}
            </ProductStatus>
          </GCol>
        </GRow>
        <GRow>
          <GCol xs={12} sm={12} md={8} lg={8}>
            <ProductName>{name}</ProductName>
          </GCol>
          <GCol xs={12} sm={12} md={4} lg={4}>
            <ProductPrice>{price}</ProductPrice>
          </GCol>
        </GRow>
      </Grid>
    </ProductWrapper>
  )
}

Product.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  isExclusive: PropTypes.bool.isRequired,
  isSale: PropTypes.bool.isRequired
}

export default Product
