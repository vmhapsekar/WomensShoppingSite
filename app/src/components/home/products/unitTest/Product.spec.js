import React from 'react'
import Product from '../Product'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'

describe('<Product />', () => {
  test('Product with Exclusive Status', () => {
    const initialProps = {
      image: 'product-1.jpg',
      name: 'Striped shirt',
      price: "$18.88",
      isExclusive: true,
      isSale: false
    }
    const component = shallow(<Product {...initialProps} />)
    expect(component).toBeDefined()
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  test('Product with Sale Status', () => {
    const initialProps = {
      image: 'product-1.jpg',
      name: 'Striped shirt',
      price: "$18.88",
      isExclusive: false,
      isSale: true
    }
    const component = shallow(<Product {...initialProps} />)
    expect(component).toBeDefined()
    expect(shallowToJson(component)).toMatchSnapshot()
  })
})
