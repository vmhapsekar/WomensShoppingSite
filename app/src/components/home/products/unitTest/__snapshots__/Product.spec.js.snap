// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> Product with Exclusive Status 1`] = `
<styled.div>
  <Grid
    fluid={true}
  >
    <GRow>
      <GCol
        xs={12}
      >
        <styled.img
          src="http://localhost:9999/images/product-1.jpg"
        />
      </GCol>
    </GRow>
    <GRow>
      <GCol
        lg={5}
        md={5}
        sm={12}
        xs={12}
      >
        <styled.div
          isExclusive={true}
          isSale={false}
        >
          Exclusive
        </styled.div>
      </GCol>
    </GRow>
    <GRow>
      <GCol
        lg={8}
        md={8}
        sm={12}
        xs={12}
      >
        <styled.div>
          Striped shirt
        </styled.div>
      </GCol>
      <GCol
        lg={4}
        md={4}
        sm={12}
        xs={12}
      >
        <styled.h2>
          $18.88
        </styled.h2>
      </GCol>
    </GRow>
  </Grid>
</styled.div>
`;

exports[`<Product /> Product with Sale Status 1`] = `
<styled.div>
  <Grid
    fluid={true}
  >
    <GRow>
      <GCol
        xs={12}
      >
        <styled.img
          src="http://localhost:9999/images/product-1.jpg"
        />
      </GCol>
    </GRow>
    <GRow>
      <GCol
        lg={5}
        md={5}
        sm={12}
        xs={12}
      >
        <styled.div
          isExclusive={false}
          isSale={true}
        >
          Sale
        </styled.div>
      </GCol>
    </GRow>
    <GRow>
      <GCol
        lg={8}
        md={8}
        sm={12}
        xs={12}
      >
        <styled.div>
          Striped shirt
        </styled.div>
      </GCol>
      <GCol
        lg={4}
        md={4}
        sm={12}
        xs={12}
      >
        <styled.h2>
          $18.88
        </styled.h2>
      </GCol>
    </GRow>
  </Grid>
</styled.div>
`;
